core/src/main/java/org/apache/iceberg/rest/auth/OAuth2Manager.java

```diff
@@ -167,6 +167,10 @@ public void close() {
     }
   }
 
+  /**
+   * @deprecated since 1.10.0, will be removed in 1.11.0.
```

Comment:
what's the alternative impl that people should be switching to? I think part of this PR we should be adding a new method that returns the generic session cache and switch `sessionCache` in this class to use it